src/services/importService.ts

```diff
@@ -0,0 +1,186 @@
+         IV1Project, IAssetMetadata, IRegion, RegionType, AssetType, AssetState } from "../models/applicationState";
+import { AssetService } from "./assetService";
+import MD5 from "md5.js";
+import { randomIntInRange } from "../common/utils";
+import TagColors from "../react/components/common/tagsInput/tagColors.json";
+// const TagColors = require("../react/components/common/tagsInput/tagColors.json");
+
+/**
+ * Functions required for an import service
+ * @member convertV1 - Converts a v1 project to v2 project
+ * @member generateAssets - Generates v2 assets based on v1 project file
+ * @member generateConnections - Generates v2 connections based on location of v1 project file
+ */
+export interface IImportService {
+    convertV1(project: any): Promise<IProject>;
+}
+
+/**
+ * @name - Import Service
+ * @description - Functions for importing v1 projects to v2 application
+ */
+// change any to { content, file } object eventually
+export default class ImportService implements IImportService {
+    public convertV1(project: any): Promise<IProject> {
+        return new Promise<IProject>((resolve, reject) => {
+            let originalProject: IV1Project;
+            let convertedProject: IProject;
+            let connections: IConnection[];
+            let parsedTags: ITag[];
+            let generatedAssetMetadata: IAssetMetadata[];
+
+            originalProject = JSON.parse(project.content);
+            parsedTags = this.parseTags(originalProject);
+
+            connections = this.generateConnections(project);
+            generatedAssetMetadata = this.generateAssets(project);
+
+            // map v1 values to v2 values
+            convertedProject = {
+                id: shortid.generate(),
+                name: project.file.name.split(".")[0],
+                version: "v1-to-v2",
+                securityToken: `${project.file.name.split(".")[0]} Token`,
+                description: "Converted V1 Project",
+                tags: parsedTags,
+                sourceConnection: connections[0],
+                targetConnection: connections[1],
+                exportFormat: null,
+                videoSettings: {
+                    frameExtractionRate: 15,
+                },
+                autoSave: true,
+                assets: {},
+            };
+
+            const assetService = new AssetService(convertedProject);
+
+            const saveAssets = generatedAssetMetadata.map((assetMetadata) => {
+                return assetService.save(assetMetadata);
+            });
+
+            try {
+                Promise.all(saveAssets);
+            } catch (e) {
+                reject(e);
+            }
+
+            resolve(convertedProject);
+        });
+    }
+
+    /**
+     * Generate connections from v1 project file location
+     * @param project - V1 Project Content and File Information
+     */
+    private generateConnections(project: any): IConnection[] {
+        const sourceConnection: IConnection = {
```

Comment:
Let's rename the connection to something memorable based on the project file name.